pandas/core/frame.py

```diff
@@ -3392,7 +3392,7 @@ def _ensure_valid_index(self, value):
         passed value.
         """
         # GH5632, make sure that we are a Series convertible
-        if not len(self.index) and is_list_like(value):
+        if not len(self.index) and is_list_like(value) and len(value):
```

Comment:
put the len(value) check before is_list_like

Reply:
That breaks a lot of other tests, `is_list_like` check is required before we do a len based checks, otherwise there are tests where obj passed will give `TypeError`